packages/core/src/utils/make-namespaced.ts

```diff
@@ -163,74 +163,75 @@ export function makeNamespacedInput(input: SolcInput, output: SolcOutput, solcVe
       }
     }
 
-    const modifiedSource = tryRemoveNonStructNatSpec(getModifiedSource(orig, modifications), solcVersion);
-
-    modifiedSources[sourcePath] = { ...source, content: modifiedSource };
+    modifiedSources[sourcePath] = { ...source, content: getModifiedSource(orig, modifications) };
   }
 
   return { ...input, sources: modifiedSources, settings: { ...input.settings, outputSelection: OUTPUT_SELECTION } };
 }
 
 /**
- * If we have the compiler version available and Slang is supported for the current platform and compiler version,
- * use Slang to parse and remove all NatSpec comments that do not precede a struct definition and return the modified content.
+ * Attempts to remove all NatSpec comments that do not precede a struct definition from the input source contents.
+ * Directly modifies the input source contents.
+ *
+ * If the solc version is not supported by the parser, the original content is kept.
+ *
+ * @param solcInput Solc input.
+ * @param solcVersion The version of the solc compiler that was originally used to compile the input.
+ */
+export async function trySanitizeNatSpec(solcInput: SolcInput, solcVersion: string) {
+  for (const [sourcePath, source] of Object.entries(solcInput.sources)) {
+    if (source.content !== undefined) {
+      solcInput.sources[sourcePath].content = await tryRemoveNonStructNatSpec(source.content, solcVersion);
+    }
+  }
+}
```

Comment:
Accordingly

```suggestion
export async function trySanitizeNatSpec(solcInput: SolcInput, solcVersion: string): Record<string, { content?: string }> {
  for (const [sourcePath, source] of Object.entries(solcInput.sources)) {
    if (source.content !== undefined) {
      solcInput.sources[sourcePath].content = await tryRemoveNonStructNatSpec(source.content, solcVersion);
    }
  }
  
  return solcInput;
}
```

I may be wrong with the types

Reply:
Reimplemented along with using the returned input in the callers.